jbmc/unit/java-testing-utils/require_goto_statements.cpp

```diff
@@ -295,6 +294,54 @@ const code_declt &require_goto_statements::require_declaration_of_name(
   throw no_decl_found_exceptiont(variable_name.c_str());
 }
 
+static bool is_allocate(const exprt &expr)
+{
+  auto side_effect_expr = expr_try_dynamic_cast<side_effect_exprt>(expr);
+  return side_effect_expr && side_effect_expr->get_statement() == ID_allocate;
+}
+
+static const irep_idt &
+get_name_of_symbol_with_struct_component_ptr_type(
+  const std::vector<codet> &entry_point_instructions,
+  const code_assignt &assignment)
+{
+  // We are expecting that the resulting statement is of the form:
+  //   malloc_site->(@superclass_name if given.)field = malloc_site$0;
+  // where the symbol on the right hand side is a symbol that either
+  // 1. has previously been dynamically allocated like this:
+  //      malloc_site$0 = ALLOCATE(struct java.lang.Object, 4ul, false);
+  // 2. has not
+
```

Comment:
I'll expand that comment to make things clearer. It should say that we're expecting a symbol has been assigned to it, and it's that symbol which we return. In the short term it doesn't matter which symbol we use as they both have the same type, but we return the symbol and some tests continue to use it for other things, and in case 2 they need the second symbol.